src/persistQueryClient/index.ts

```diff
@@ -1,142 +1,2 @@
-  extends PersistQueryClienRootOptions {
-  /** The options passed to the dehydrate function */
-  dehydrateOptions?: DehydrateOptions
-}
-
-export interface PersistQueryClientOptions
-  extends PersistedQueryClientRestoreOptions,
-    PersistedQueryClientSaveOptions,
-    PersistQueryClienRootOptions {}
-
-/**
- * Restores persisted data to the QueryCache
- *  - data obtained from persister.restoreClient
- *  - data is hydrated using hydrateOptions
- * If data is expired, busted, empty, or throws, it runs persister.removeClient
- */
-export async function persistQueryClientRestore({
-  queryClient,
-  persister,
-  maxAge = 1000 * 60 * 60 * 24,
-  buster = '',
-  hydrateOptions,
-}: PersistedQueryClientRestoreOptions) {
-  if (typeof window !== 'undefined') {
-    try {
-      const persistedClient = await persister.restoreClient()
-
-      if (persistedClient) {
-        if (persistedClient.timestamp) {
-          const expired = Date.now() - persistedClient.timestamp > maxAge
-          const busted = persistedClient.buster !== buster
-          if (expired || busted) {
-            persister.removeClient()
-          } else {
-            hydrate(queryClient, persistedClient.clientState, hydrateOptions)
-          }
-        } else {
-          persister.removeClient()
-        }
-      }
-    } catch (err) {
-      getLogger().error(err)
-      getLogger().warn(
-        'Encountered an error attempting to restore client cache from persisted location. As a precaution, the persisted cache will be discarded.'
-      )
-      persister.removeClient()
-    }
-  }
-}
-
-/**
- * Persists data from the QueryCache
- *  - data dehydrated using dehydrateOptions
- *  - data is persisted using persister.persistClient
- */
-export async function persistQueryClientSave({
-  queryClient,
-  persister,
-  buster = '',
-  dehydrateOptions,
-}: PersistedQueryClientSaveOptions) {
-  if (typeof window !== 'undefined') {
-    const persistClient: PersistedClient = {
-      buster,
-      timestamp: Date.now(),
-      clientState: dehydrate(queryClient, dehydrateOptions),
-    }
-
-    await persister.persistClient(persistClient)
-  }
-}
-
-/**
- * Subscribe to QueryCache updates (for persisting)
- * @returns an unsubscribe function (to discontinue monitoring)
- */
-export function persistQueryClientSubscribe(
-  props: PersistedQueryClientSaveOptions
-) {
-  return props.queryClient.getQueryCache().subscribe(() => {
-    persistQueryClientSave(props)
-  })
-}
-
-/**
- * Restores persisted data to QueryCache and persists further changes.
- * (Retained for backwards compatibility)
- */
-export async function persistQueryClient(props: PersistQueryClientOptions) {
-  if (typeof window !== 'undefined') {
-    // Attempt restore
-    await persistQueryClientRestore(props)
-
-    // Subscribe to changes in the query cache to trigger the save
-    return persistQueryClientSubscribe(props)
-  }
-}
+export * from './persist'
+export * from './PersistQueryClientProvider'
```

Comment:
I am not sure where to put the `PersistQueryClientProvider`. It is both react specific as well as specific for the plugin. If I put it to `react`, it will likely wind up in the main bundle (even though treeshakeable). If we keep it here, the `persistQueryClient` plugin is dependent on React...

Reply:
Don't have a solution, just wanted to say I had/have the same problem with the hydrate stuff. :)

Reply:
hydration was moved to the core, but I really don't want to move this plugin to the core. I guess I'll leave it here for now, until someone complains that they want to use the plugin without react or so 😅